crates/biome_js_syntax/src/expr_ext.rs

```diff
@@ -284,6 +290,22 @@ impl JsBinaryExpression {
             Ok(false)
         }
     }
+
+    pub fn get_node_position(&self, node: &JsSyntaxNode) -> Option<BinaryExpressionNodePosition> {
```

Comment:
I have some reservations about this function. Here's why:
- the `Option` is used because our APIs return `Result` in case of broken syntax, but at end of the function you return `None`, like a node isn't either right or left. That seems wrong to me.
- using a text to compare nodes isn't reliable. You should use their range. The range is fool proof and you can always say where the node is.
- some documentation will help us to help you with the implementation

Reply:
Here some context about the method:

In use_shortand_operator and no_misrefactored_shorthand rules, i need to know the position of variable on a binary expression, so given the expression `a = a + b`, i want to know if variable `a` is on left or right side of the expression.

If you think this don't make sense, i can find other simple way to do that (maybe i need some help from you to figure out how 😄).

About your reservations:

1. If the argument node: &JsSyntaxNode don't match none of sides, we can't return either Left or Right
2. Can you send me some examples how to use range to compare ? It will be very helpful
3. Sure ✍🏻

Reply:
> If the argument node: &JsSyntaxNode don't match none of sides, we can't return either Left or Right

That's the thing, which I find incorrect. 

I think this function **shouldn't** stay here, but it should be moved inside the rule file and kept there. If we keep it here, technically `node` can also be a node that doesn't belong to the AST root you're querying, and this is an incorrect assumption IMHO.

Instead, we should assume that `JsBinaryExpression` and the `node` belong to the same root. If they do, then we can using their ranges to always return a value.

You can get familiar with ranges by looking at the [Syntax tab of the playground](https://biomejs.dev/playground/?indentStyle=space&quoteStyle=single&jsxQuoteStyle=single&lintRules=all&code=YQAgAD0AIABhACAAKwAgAGIA). Each `SyntaxNode` has a method called `range()` and `trimmed_range()`. They return a `TextRange`, and this type has a `start()` method and `end()` method. You can use the `left` and `right` ranges and compare them with `node.trimmed_range`.

Reply:
Thank you for the explanation, I'll keep it inside the rule file.

Let me ask you one more thing
I need this function for 2 rules and didn't want it to be duplicated

What do you suggest here? Export the function from one rule file and then import in other ?

Reply:
You can have a module in `biome_js_analyze/src/` and call it with a name you see fit. We do the same with `react.rs`, keeping functions around react logic.